src/net.cpp

```diff
@@ -1625,7 +1626,7 @@ void CConnman::ThreadOpenConnections()
         {
             LOCK(cs_vNodes);
             BOOST_FOREACH(CNode* pnode, vNodes) {
-                if (!pnode->fInbound) {
+                if (!pnode->fInbound && !pnode->fAddnode) {
```

Comment:
That was the intent. At least the idea was to make the automatic and addnode connections operate largely independent of each other, not auto-connecting to amazon because you've got an addnode to amazon seems odd.

At least my perspective on the netgroup limitation is that with your limited 8 outbound slots you don't want to harm your partition resistance by spending multiple of them on a single netgroup.  That no longer applies when addnodes are not consuming your outbound slots, just like it doesn't apply to inbound peers. (though inbound has a stronger additional reason: they're attacker controlled).